monkey/infection_monkey/network/network_scanner.py

```diff
@@ -1,22 +1,51 @@
+import sys
 import time
+import logging
 
-from common.network.network_range import *
+from common.network.network_range import NetworkRange
 from infection_monkey.config import WormConfiguration
 from infection_monkey.network.info import local_ips, get_interfaces_ranges
 from infection_monkey.model import VictimHost
 from infection_monkey.network import TcpScanner, PingScanner
 
+if sys.platform.startswith("win"):
+    from multiprocessing.dummy import Pool
+else:
+    from multiprocessing import Pool
+
 __author__ = 'itamar'
 
 LOG = logging.getLogger(__name__)
 
 SCAN_DELAY = 0
+ITERATION_BLOCK_SIZE = 5
+
+
+def generate_victims(net_ranges, chunk_size):
+    """
+    Generates VictimHosts in chunks from all the netranges
+    :param net_ranges: Iterable of network ranges
+    :param chunk_size: Maximum size of each chunk
+    :return:
+    """
+    chunk = []
+    for net_range in net_ranges:
+        for address in net_range:
+            if hasattr(net_range, 'domain_name'):
```

Comment:
You want to move the entire generator there, along with the filtering?

Reply:
No :) Generator is one thing, creating a VictimHost object is another. The generator logic is very cohesive with the `network_scanner` right now, it will be a bad idea to separate them.